mode/src/processing/mode/android/SDKDownloader.java

```diff
@@ -623,6 +637,111 @@ public boolean cancelled() {
   public AndroidSDK getSDK() {
     return sdk;
   }
+
+  private void addPackage(GridBagConstraints gc, int gridy, JPanel packagesPanel,
+                          String packageName, String packageVersion){
+    gc.gridx = 0; gc.gridy = gridy;
+    gc.weightx = 0.5;
+    JLabel packageLabel = new JLabel(packageName);
+    packagesPanel.add(packageLabel,gc);
+    JLabel versionLabel = new JLabel(packageVersion);
+    gc.gridx = 1; gc.gridy = gridy;
+    gc.weightx = 1;
+    packagesPanel.add(versionLabel,gc);
+  }
+
+  private void createInitLayout(SDKUrlHolder downloadUrls) {
+    Container outer = getContentPane();
+    outer.removeAll();
+    final SDKUrlHolder urlHolder = downloadUrls;
+
+    outer.setLayout(new BorderLayout());
+
+    JPanel mainPanel = new JPanel();
+    mainPanel.setPreferredSize(Toolkit.zoom(250,225));
+    outer.add(mainPanel,BorderLayout.EAST);
+
+    //List Packages to be downloaded here:
+    JPanel packagesPanel = new JPanel();
+    packagesPanel.setPreferredSize(Toolkit.zoom(250,125));
+    packagesPanel.setAlignmentX(RIGHT_ALIGNMENT);
+    Border border = BorderFactory.createLineBorder(Color.black,1,true);
+    packagesPanel.setBorder(BorderFactory.createTitledBorder(border,"Packages to download: "));
+    packagesPanel.setLayout(new GridBagLayout());
+
+    mainPanel.add(packagesPanel);
+
+    GridBagConstraints gc = new GridBagConstraints();
+    gc.anchor = NORTHWEST;
+    gc.weighty = 0.5;
+
+    addPackage(gc,0,packagesPanel,"SDK Platform: ",downloadUrls.platformVersion);
+    addPackage(gc,1,packagesPanel,"SDK PlatformTools: ",downloadUrls.platformToolsVersion);
+    addPackage(gc,2,packagesPanel,"Android Build Tools: ",downloadUrls.buildToolsVersion);
+    addPackage(gc,3,packagesPanel,"Android Tools: ",downloadUrls.toolsVersion);
+    addPackage(gc,4,packagesPanel,"Android Emulator: ",downloadUrls.emulatorVersion);
```

Comment:
This indicates that during the installation process of the AndroidSDK, Processing will install the Android SDK as well as a standard emulator, correct? Potentially we could come back later to add extra details about the emulator. We can address this later after milestone 2 is done. Thoughts? @DeerajTheepshi

Reply:
I'm not sure about what you mean by adding extra details to emulator @kfrajer

Reply:
Sorry, I was not clear. I will send you a message in about 12hours about this. I would like to discuss about the user experience during the Android SDK installation as well as the emulator. Please mark this as resolved and i will start the discussion in the evol doc. @DeerajTheepshi